source/common/http/status.cc

```diff
@@ -21,6 +21,8 @@ absl::string_view statusCodeToString(StatusCode code) {
     return "PrematureResponseError";
   case StatusCode::CodecClientError:
     return "CodecClientError";
+  default:
+    RELEASE_ASSERT(false, "Invalid StatusCode value");
```

Comment:
I think this should also be a normal ASSERT, there's a guarantee `StatusCode` will have the above possibilities.

Reply:
compiling with a non-debug build, an `ASSERT` still causes the compiler to not believe the control path returns

```
C:\_eb\execroot\envoy\source\common\http\status.cc(27) : error C2220: the following warning is treated as an error
C:\_eb\execroot\envoy\source\common\http\status.cc(27) : warning C4715: 'Envoy::Http::`anonymous namespace'::statusCodeToString': not all control paths return a value
```